simple-page-ordering.php

```diff
@@ -168,10 +168,26 @@ public static function load_edit_screen() {
 					'sort_by_order_link',
 				)
 			);
+			add_action( 'pre_get_posts', array( __CLASS__, 'filter_query' ) );
 			add_action( 'wp', array( __CLASS__, 'wp' ) );
 			add_action( 'admin_head', array( __CLASS__, 'admin_head' ) );
 		}
 
+		/**
+		 * This is to enable pagination.
+		 *
+		 * @param WP_Query $query The WP_Query instance (passed by reference).
+		 */
+		public static function filter_query( $query ) {
+			$is_simple_page_ordering = isset( $_GET['id'] ) ? 'simple-page-ordering' === sanitize_text_field( wp_unslash( $_GET['id'] ) ) : false;
+
+			if ( ! $is_simple_page_ordering ) {
+				return;
+			}
+
+			$query->set( 'posts_per_page', -1 );
```

Comment:
I'm unclear how this works, it appears to remove paging but forces it. Are you able to explain to me what I am missing?

Reply:
Sure. Comparing the issue with how clicking "All" works, the reason why "Sort by Order" does not work is because it adds a LIMIT to the query here:

https://github.com/WordPress/wordpress-develop/blob/d8206d2167f2c082258f31db163289bae3682e06/src/wp-includes/class-wp-query.php#L2745

To avoid that and make the "Sort By Order" work similarly to "All", we set 'nopaging' to `true` here:
https://github.com/WordPress/wordpress-develop/blob/d8206d2167f2c082258f31db163289bae3682e06/src/wp-includes/class-wp-query.php#L1988 by setting `posts_per_page` to `-1`.